bundles/com.espressif.idf.core/src/com/espressif/idf/core/SystemExecutableFinder.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright 2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core;
+
+import java.io.File;
+import java.lang.reflect.Method;
+
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.filesystem.IFileStore;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Platform;
+
+import com.espressif.idf.core.util.StringUtil;
+
+public class SystemExecutableFinder implements ExecutableFinder
+{
+
+	private SystemWrapper systemWrapper;
+
+	public SystemExecutableFinder()
+	{
+		this.systemWrapper = new DefaultSystemWrapper();
+	}
+
+	public SystemExecutableFinder(SystemWrapper systemWrapper)
+	{
+		this.systemWrapper = systemWrapper;
+	}
```

Comment:
I decided to not use a singleton for this utility class because it can be beneficial to not have it singleton for testing and maintenance purposes, also modern JVMs are designed to handle object creation and destruction efficiently, so creating new instances of a stateless utility class should not cause any significant performance issues. 

Also not use the static factory method because the constructors here are clean and I see benefits in the future from being able to have subclasses of this class